Lab 3: Association rules

Jorge Angon, Valerie Jimenez


In [ ]:
import pandas as pd
import numpy as np
from mlxtend.frequent_patterns import apriori, association_rules

df = pd.read_csv("Supermart.csv")
df.head()
#call libraries that are needed ands read the csv file and display it


/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)


,Order ID,Customer Name,Category,Sub Category,City,Order Date,Region,Sales,Discount,Profit,State
0,OD1,Harish,Oil & Masala,Masalas,Vellore,11-08-2017,North,1254,0.12,401.28,Tamil Nadu
1,OD2,Sudha,Beverages,Health Drinks,Krishnagiri,11-08-2017,South,749,0.18,149.80,Tamil Nadu
2,OD3,Hussain,Food Grains,Atta & Flour,Perambalur,06-12-2017,West,2360,0.21,165.20,Tamil Nadu
3,OD4,Jackson,Fruits & Veggies,Fresh Vegetables,Dharmapuri,10-11-2016,South,896,0.25,89.60,Tamil Nadu
4,OD5,Ridhesh,Food Grains,Organic Staples,Ooty,10-11-2016,South,2355,0.26,918.45,Tamil Nadu


/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)


In [ ]:
print(df.isnull().sum()) #Shows missing values. output shows that there are no missing values.

Order ID         0
Customer Name    0
Category         0
Sub Category     0
City             0
Order Date       0
Region           0
Sales            0
Discount         0
Profit           0
State            0
dtype: int64


/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packag

In [ ]:
# Make a new table called "products"
# Each row is a basket of Sub Category items

df["Order Date"] = pd.to_datetime(df["Order Date"], format="mixed")
df["OrderMonth"] = df["Order Date"].dt.to_period("M")

df = (
    df.groupby(["Customer Name", "OrderMonth"])["Sub Category"]
      .apply(lambda x: ",".join(sorted(set(x))))
      .reset_index(name="products")
)

# Keep only the column we need
df = df[["products"]]
df.head()


/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packag

,products
0,Chicken
1,"Health Drinks,Mutton"
2,"Fresh Fruits,Masalas,Spices"
3,"Edible Oil & Ghee,Mutton,Spices"
4,"Edible Oil & Ghee,Health Drinks,Organic Staples"


/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)


In [ ]:
df.shape #Shows the rows and columns


/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packag

(2268, 1)

/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)


In [ ]:
# Turns into a list of items
data = list(df["products"].apply(lambda x: x.split(",")))
data


/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packag

[['Chicken'],
 ['Health Drinks', 'Mutton'],
 ['Fresh Fruits', 'Masalas', 'Spices'],
 ['Edible Oil & Ghee', 'Mutton', 'Spices'],
 ['Edible Oil & Ghee', 'Health Drinks', 'Organic Staples'],
 ['Chocolates', 'Cookies', 'Rice', 'Spices'],
 ['Chicken', 'Chocolates', 'Cookies', 'Dals & Pulses', 'Mutton'],
 ['Chocolates', 'Fresh Fruits', 'Health Drinks'],
 ['Atta & Flour',
  'Chocolates',
  'Dals & Pulses',
  'Health Drinks',
  'Mutton',
  'Rice',
  'Soft Drinks'],
 ['Atta & Flour',
  'Chocolates',
  'Dals & Pulses',
  'Fish',
  'Masalas',
  'Soft Drinks'],
 ['Cookies', 'Dals & Pulses'],
 ['Organic Vegetables'],
 ['Biscuits', 'Chocolates'],
 ['Breads & Buns', 'Chicken', 'Fresh Fruits'],
 ['Biscuits', 'Health Drinks', 'Soft Drinks'],
 ['Chocolates', 'Fresh Fruits'],
 ['Breads & Buns', 'Dals & Pulses', 'Fresh Fruits', 'Organic Fruits'],
 ['Fresh Vegetables', 'Soft Drinks', 'Spices'],
 ['Breads & Buns'],
 ['Cookies',
  'Fish',
  'Noodles',
  'Organic Fruits',
  'Organic Vegetables',
  'Spices'],


/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)


In [ ]:
# Replaces the true ands false and makes turns them into 1 or 0(one-hot encoding)
# Columns = items, rows = baskets

from mlxtend.preprocessing import TransactionEncoder
a = TransactionEncoder()
a_data = a.fit(data).transform(data)
df = pd.DataFrame(a_data, columns=a.columns_)

# Turn False/True into 0/1
df = df.replace(False, 0)
df = df.replace(True, 1)

df


/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packag

,Atta & Flour,Biscuits,Breads & Buns,Cakes,Chicken,Chocolates,Cookies,Dals & Pulses,Edible Oil & Ghee,Eggs,...,Health Drinks,Masalas,Mutton,Noodles,Organic Fruits,Organic Staples,Organic Vegetables,Rice,Soft Drinks,Spices
0,0,0,0,0,1,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
1,0,0,0,0,0,0,0,0,0,0,...,1,0,1,0,0,0,0,0,0,0
2,0,0,0,0,0,0,0,0,0,0,...,0,1,0,0,0,0,0,0,0,1
3,0,0,0,0,0,0,0,0,1,0,...,0,0,1,0,0,0,0,0,0,1
4,0,0,0,0,0,0,0,0,1,0,...,1,0,0,0,0,1,0,0,0,0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
2263,1,0,0,0,0,0,1,0,1,1,...,0,1,0,1,0,0,0,0,0,0
2264,1,0,1,0,0,1,0,0,1,0,...,0,0,0,1,0,0,0,0,0,1
2265,1,0,0,0,0,0,0,0,1,1,...,0,1,0,0,0,0,0,0,0,0
2266,1,1,0,0,0,1,0,0,1,0,...,1,1,0,1,0,1,0,0,0,0


/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)


In [ ]:
# Finds the items groups that show up often
df = apriori(df, min_support=0.05, use_colnames=True)
df


/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packag

,support,itemsets
0,0.145062,(Atta & Flour)
1,0.184303,(Biscuits)
2,0.194885,(Breads & Buns)
3,0.178571,(Cakes)
4,0.138889,(Chicken)
5,0.196649,(Chocolates)
6,0.201940,(Cookies)
7,0.140212,(Dals & Pulses)
8,0.177249,(Edible Oil & Ghee)
9,0.151675,(Eggs)


/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)


In [ ]:
# confidence rule.
df_ar = association_rules(df, metric="confidence", min_threshold=0.30)
df_ar


/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packag

,antecedents,consequents,antecedent support,consequent support,support,confidence,lift,representativity,leverage,conviction,zhangs_metric,jaccard,certainty,kulczynski
0,(Breads & Buns),(Health Drinks),0.194885,0.270282,0.059965,0.307692,1.138411,1.0,0.007291,1.054037,0.151013,0.147987,0.051267,0.264776
1,(Breads & Buns),(Soft Drinks),0.194885,0.259700,0.064374,0.330317,1.271916,1.0,0.013762,1.105448,0.265533,0.164972,0.095389,0.289097
2,(Edible Oil & Ghee),(Health Drinks),0.177249,0.270282,0.053792,0.303483,1.122836,1.0,0.005885,1.047666,0.132966,0.136618,0.045498,0.251252
3,(Masalas),(Health Drinks),0.181658,0.270282,0.059083,0.325243,1.203345,1.0,0.009984,1.081452,0.206494,0.150393,0.075318,0.271920
4,(Organic Staples),(Health Drinks),0.150353,0.270282,0.050265,0.334311,1.236896,1.0,0.009627,1.096184,0.225416,0.135714,0.087744,0.260141
5,(Soft Drinks),(Health Drinks),0.259700,0.270282,0.081570,0.314092,1.162088,1.0,0.011377,1.063871,0.188410,0.181908,0.060036,0.307943
6,(Health Drinks),(Soft Drinks),0.270282,0.259700,0.081570,0.301794,1.162088,1.0,0.011377,1.060289,0.191142,0.181908,0.056861,0.307943
7,(Spices),(Health Drinks),0.174603,0.270282,0.052910,0.303030,1.121163,1.0,0.005718,1.046986,0.130929,0.134983,0.044878,0.249394


/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)


In [ ]:
# Lift rules
df_ar = association_rules(df, metric="lift", min_threshold=1)
df_ar


/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packag

,antecedents,consequents,antecedent support,consequent support,support,confidence,lift,representativity,leverage,conviction,zhangs_metric,jaccard,certainty,kulczynski
0,(Biscuits),(Health Drinks),0.184303,0.270282,0.053792,0.291866,1.079857,1.0,0.003978,1.030480,0.090660,0.134213,0.029578,0.245444
1,(Health Drinks),(Biscuits),0.270282,0.184303,0.053792,0.199021,1.079857,1.0,0.003978,1.018375,0.101342,0.134213,0.018043,0.245444
2,(Soft Drinks),(Biscuits),0.259700,0.184303,0.054233,0.208829,1.133070,1.0,0.006369,1.030999,0.158641,0.139140,0.030067,0.251543
3,(Biscuits),(Soft Drinks),0.184303,0.259700,0.054233,0.294258,1.133070,1.0,0.006369,1.048967,0.143977,0.139140,0.046681,0.251543
4,(Breads & Buns),(Health Drinks),0.194885,0.270282,0.059965,0.307692,1.138411,1.0,0.007291,1.054037,0.151013,0.147987,0.051267,0.264776
5,(Health Drinks),(Breads & Buns),0.270282,0.194885,0.059965,0.221860,1.138411,1.0,0.007291,1.034665,0.166616,0.147987,0.033504,0.264776
6,(Soft Drinks),(Breads & Buns),0.259700,0.194885,0.064374,0.247878,1.271916,1.0,0.013762,1.070457,0.288781,0.164972,0.065820,0.289097
7,(Breads & Buns),(Soft Drinks),0.194885,0.259700,0.064374,0.330317,1.271916,1.0,0.013762,1.105448,0.265533,0.164972,0.095389,0.289097
8,(Cakes),(Health Drinks),0.178571,0.270282,0.051587,0.288889,1.068842,1.0,0.003323,1.026166,0.078410,0.129856,0.025498,0.239877
9,(Health Drinks),(Cakes),0.270282,0.178571,0.051587,0.190865,1.068842,1.0,0.003323,1.015193,0.088264,0.129856,0.014966,0.239877


/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)


In [ ]:
# Conviction rules
df_ar = association_rules(df, metric="conviction", min_threshold=1.0)
df_ar


/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packag

,antecedents,consequents,antecedent support,consequent support,support,confidence,lift,representativity,leverage,conviction,zhangs_metric,jaccard,certainty,kulczynski
0,(Health Drinks),(Biscuits),0.270282,0.184303,0.053792,0.199021,1.079857,1.0,0.003978,1.018375,0.101342,0.134213,0.018043,0.245444
1,(Biscuits),(Health Drinks),0.184303,0.270282,0.053792,0.291866,1.079857,1.0,0.003978,1.030480,0.090660,0.134213,0.029578,0.245444
2,(Biscuits),(Soft Drinks),0.184303,0.259700,0.054233,0.294258,1.133070,1.0,0.006369,1.048967,0.143977,0.139140,0.046681,0.251543
3,(Soft Drinks),(Biscuits),0.259700,0.184303,0.054233,0.208829,1.133070,1.0,0.006369,1.030999,0.158641,0.139140,0.030067,0.251543
4,(Breads & Buns),(Health Drinks),0.194885,0.270282,0.059965,0.307692,1.138411,1.0,0.007291,1.054037,0.151013,0.147987,0.051267,0.264776
5,(Health Drinks),(Breads & Buns),0.270282,0.194885,0.059965,0.221860,1.138411,1.0,0.007291,1.034665,0.166616,0.147987,0.033504,0.264776
6,(Breads & Buns),(Soft Drinks),0.194885,0.259700,0.064374,0.330317,1.271916,1.0,0.013762,1.105448,0.265533,0.164972,0.095389,0.289097
7,(Soft Drinks),(Breads & Buns),0.259700,0.194885,0.064374,0.247878,1.271916,1.0,0.013762,1.070457,0.288781,0.164972,0.065820,0.289097
8,(Health Drinks),(Cakes),0.270282,0.178571,0.051587,0.190865,1.068842,1.0,0.003323,1.015193,0.088264,0.129856,0.014966,0.239877
9,(Cakes),(Health Drinks),0.178571,0.270282,0.051587,0.288889,1.068842,1.0,0.003323,1.026166,0.078410,0.129856,0.025498,0.239877


/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packages/jupyter_client/session.py:203: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.utcnow().replace(tzinfo=utc)
/usr/local/lib/python3.12/dist-packag

The Supermart dataset contains orders placed by customers on a grocery app. We grouped the data by customers and months, then converted the column into a list. We then transformed the list with one-hot encoding to represent each transaction. We viewed the calculations of support, confidence, lift, and conviction to view any purchase behaviors or patterns with each product/purchase. The most frequent item that was purchased was healthy drinks with a value of .27. The probability of organic staples and health drinks being purchased was at a high confidence level of 33 and implies that if a person buys organic staples there is a 33% chance that a healthy drink will be purchased. We can see a positive correlation with soft drinks and bread being purchased with a lift of 1.27 and are more likely to be bough together. By looking at the conviction rule, we can see that bread and soft drinks are positively associated, and both of these items directly impact each other with a conviction of 1.1.